crates/autopilot/src/database/auction.rs

```diff
@@ -61,14 +62,15 @@ impl QuoteStoring for Postgres {
         &self,
         params: QuoteSearchParameters,
         expiration: DateTime<Utc>,
+        signing_scheme: &QuoteSigningScheme,
     ) -> Result<Option<(QuoteId, QuoteData)>> {
         let _timer = super::Metrics::get()
             .database_queries
             .with_label_values(&["find_quote"])
             .start_timer();
 
         let mut ex = self.0.acquire().await?;
-        let params = create_db_search_parameters(params, expiration);
+        let params = create_db_search_parameters(params, expiration, signing_scheme);
```

Comment:
nit: I like how the `quote_kind` was moved out of the `QuoteSearchParameters` as it simplified code in a few places. That being said, I'm not a huge fan of having the `QuoteSigningScheme` "leak" into the database finding code (mostly because it is immediately converted to an `OrderKind` anyway, so we should be passing that down instead of a signing scheme in my opinion. I created a [commit with a suggested change](https://github.com/cowprotocol/services/tree/keep-quote-kind-in-search). What is nice is that it removes some code duplication in converting `QuoteSigningScheme -> QuoteKind`:

```diff
diff --git a/crates/autopilot/src/database/auction.rs b/crates/autopilot/src/database/auction.rs
index a23af643..b5504b9a 100644
--- a/crates/autopilot/src/database/auction.rs
+++ b/crates/autopilot/src/database/auction.rs
@@ -6,6 +6,7 @@ use database::{
         BuyTokenDestination as DbBuyTokenDestination, SellTokenSource as DbSellTokenSource,
         SigningScheme as DbSigningScheme,
     },
+    quotes::QuoteKind,
 };
 use futures::{StreamExt, TryStreamExt};
 use model::{
@@ -15,7 +16,6 @@ use model::{
         BuyTokenDestination, Order, OrderData, OrderMetadata, OrderStatus, OrderUid,
         SellTokenSource,
     },
-    quote::QuoteSigningScheme,
     signature::{Signature, SigningScheme},
 };
 use number_conversions::{big_decimal_to_big_uint, big_decimal_to_u256};
@@ -62,7 +62,7 @@ impl QuoteStoring for Postgres {
         &self,
         params: QuoteSearchParameters,
         expiration: DateTime<Utc>,
-        signing_scheme: &QuoteSigningScheme,
+        quote_kind: QuoteKind,
     ) -> Result<Option<(QuoteId, QuoteData)>> {
         let _timer = super::Metrics::get()
             .database_queries
@@ -70,7 +70,7 @@ impl QuoteStoring for Postgres {
             .start_timer();
 
         let mut ex = self.0.acquire().await?;
-        let params = create_db_search_parameters(params, expiration, signing_scheme);
+        let params = create_db_search_parameters(params, expiration, quote_kind);
         let quote = database::quotes::find(&mut ex, &params)
             .await
             .context("failed finding quote by parameters")?;
diff --git a/crates/autopilot/src/database/onchain_order_events.rs b/crates/autopilot/src/database/onchain_order_events.rs
index d4a95e16..0c74ea06 100644
--- a/crates/autopilot/src/database/onchain_order_events.rs
+++ b/crates/autopilot/src/database/onchain_order_events.rs
@@ -707,9 +707,9 @@ mod test {
 
         let signing_scheme = QuoteSigningScheme::Eip1271 {
             onchain_order: true,
-            verification_gas_limit: default_verification_gas_limit()
+            verification_gas_limit: default_verification_gas_limit(),
         };
-        
+
         let event_data_1 = EthContractEvent {
             data: ContractEvent::OrderPlacement(order_placement.clone()),
             meta: Some(EventMetadata {
@@ -730,25 +730,13 @@ mod test {
         let mut order_quoter = MockOrderQuoting::new();
         order_quoter
             .expect_find_quote()
-            .with(
-                eq(Some(quote_id_1)),
-                always(),
-                eq(signing_scheme),
-            )
-            .with(
-                eq(Some(quote_id_1)),
-                always(),
-                eq(signing_scheme),
-            )
+            .with(eq(Some(quote_id_1)), always(), eq(signing_scheme))
+            .with(eq(Some(quote_id_1)), always(), eq(signing_scheme))
             .returning(move |_, _, _| Ok(Quote::default()));
         let quote_id_2 = 6i64;
         order_quoter
             .expect_find_quote()
-            .with(
-                eq(Some(quote_id_2)),
-                always(),
-                eq(signing_scheme),
-            )
+            .with(eq(Some(quote_id_2)), always(), eq(signing_scheme))
             .returning(move |_, _, _| Err(FindQuoteError::NotFound(None)));
         let result_vec = parse_general_onchain_order_placement_data(
             &order_quoter,
diff --git a/crates/model/src/quote.rs b/crates/model/src/quote.rs
index d20daaf2..2e3324e5 100644
--- a/crates/model/src/quote.rs
+++ b/crates/model/src/quote.rs
@@ -2,7 +2,8 @@ use crate::{
     app_id::AppId,
     order::{BuyTokenDestination, OrderKind, SellTokenSource},
     signature::SigningScheme,
-    time, u256_decimal::{self, DecimalU256},
+    time,
+    u256_decimal::{self, DecimalU256},
 };
 use anyhow::bail;
 use chrono::{DateTime, Utc};
@@ -88,7 +89,7 @@ impl TryFrom<QuoteSigningDeserializationData> for QuoteSigningScheme {
             (SigningScheme::Eip1271, onchain_order, verification_gas_limit) => Ok(Self::Eip1271 {
                 onchain_order,
                 verification_gas_limit: verification_gas_limit
-                    .unwrap_or(default_verification_gas_limit()),
+                    .unwrap_or_else(default_verification_gas_limit),
             }),
             (SigningScheme::PreSign, onchain_order, None) => Ok(Self::PreSign { onchain_order }),
             (_, _, Some(_)) => {
diff --git a/crates/orderbook/src/database/quotes.rs b/crates/orderbook/src/database/quotes.rs
index 0d52d54c..b1523cff 100644
--- a/crates/orderbook/src/database/quotes.rs
+++ b/crates/orderbook/src/database/quotes.rs
@@ -1,7 +1,8 @@
 use super::Postgres;
 use anyhow::{Context, Result};
 use chrono::{DateTime, Utc};
-use model::quote::{QuoteId, QuoteSigningScheme};
+use database::quotes::QuoteKind;
+use model::quote::QuoteId;
 use shared::{
     event_storing_helpers::{create_db_search_parameters, create_quote_row},
     order_quoting::{QuoteData, QuoteSearchParameters, QuoteStoring},
@@ -36,7 +37,7 @@ impl QuoteStoring for Postgres {
         &self,
         params: QuoteSearchParameters,
         expiration: DateTime<Utc>,
-        signing_scheme: &QuoteSigningScheme,
+        quote_kind: QuoteKind,
     ) -> Result<Option<(QuoteId, QuoteData)>> {
         let _timer = super::Metrics::get()
             .database_queries
@@ -44,7 +45,7 @@ impl QuoteStoring for Postgres {
             .start_timer();
 
         let mut ex = self.pool.acquire().await?;
-        let params = create_db_search_parameters(params, expiration, signing_scheme);
+        let params = create_db_search_parameters(params, expiration, quote_kind);
         let quote = database::quotes::find(&mut ex, &params)
             .await
             .context("failed finding quote by parameters")?;
diff --git a/crates/shared/src/event_storing_helpers.rs b/crates/shared/src/event_storing_helpers.rs
index e846e481..99f81dd0 100644
--- a/crates/shared/src/event_storing_helpers.rs
+++ b/crates/shared/src/event_storing_helpers.rs
@@ -26,12 +26,8 @@ pub fn create_quote_row(data: QuoteData) -> DbQuote {
     }
 }
 
-pub fn create_db_search_parameters(
-    params: QuoteSearchParameters,
-    expiration: DateTime<Utc>,
-    signing_scheme: &QuoteSigningScheme,
-) -> DbQuoteSearchParameters {
-    let quote_kind = match signing_scheme {
+pub fn quote_kind_from_signing_scheme(scheme: &QuoteSigningScheme) -> QuoteKind {
+    match scheme {
         QuoteSigningScheme::Eip1271 {
             onchain_order: true,
             ..
@@ -40,8 +36,14 @@ pub fn create_db_search_parameters(
             onchain_order: true,
         } => QuoteKind::PreSignOnchainOrder,
         _ => QuoteKind::Standard,
-    };
+    }
+}
 
+pub fn create_db_search_parameters(
+    params: QuoteSearchParameters,
+    expiration: DateTime<Utc>,
+    quote_kind: QuoteKind,
+) -> DbQuoteSearchParameters {
     DbQuoteSearchParameters {
         sell_token: ByteArray(params.sell_token.0),
         buy_token: ByteArray(params.buy_token.0),
@@ -50,6 +52,6 @@ pub fn create_db_search_parameters(
         buy_amount: u256_to_big_decimal(&params.buy_amount),
         kind: order_kind_into(params.kind),
         expiration,
-        quote_kind: quote_kind,
+        quote_kind,
     }
 }
diff --git a/crates/shared/src/fee_subsidy.rs b/crates/shared/src/fee_subsidy.rs
index d4a8cbf2..521a0d1a 100644
--- a/crates/shared/src/fee_subsidy.rs
+++ b/crates/shared/src/fee_subsidy.rs
@@ -153,7 +153,11 @@ impl FeeParameters {
         self.subsidized_with_additional_cost(subsidy, 0.into())
     }
 
-    pub fn subsidized_with_additional_cost(&self, subsidy: &Subsidy, additional_cost: U256) -> U256 {
+    pub fn subsidized_with_additional_cost(
+        &self,
+        subsidy: &Subsidy,
+        additional_cost: U256,
+    ) -> U256 {
         // convert Eth value to f64
         let additional_cost = additional_cost.to_f64_lossy();
         let fee_in_eth = (self.gas_amount + additional_cost) * self.gas_price;
diff --git a/crates/shared/src/order_quoting.rs b/crates/shared/src/order_quoting.rs
index 4c74484e..e3af883d 100644
--- a/crates/shared/src/order_quoting.rs
+++ b/crates/shared/src/order_quoting.rs
@@ -192,7 +192,10 @@ impl Quote {
         };
 
         // THIS CANNOT MODIFY `quote.data`.
-        self.fee_amount = self.data.fee_parameters.subsidized_with_additional_cost(subsidy, verification_fee);
+        self.fee_amount = self
+            .data
+            .fee_parameters
+            .subsidized_with_additional_cost(subsidy, verification_fee);
         self
     }
 
@@ -379,7 +382,7 @@ pub trait QuoteStoring: Send + Sync {
         &self,
         parameters: QuoteSearchParameters,
         expiration: DateTime<Utc>,
-        signing_scheme: &QuoteSigningScheme,
+        quote_kind: QuoteKind,
     ) -> Result<Option<(QuoteId, QuoteData)>>;
 }
 
@@ -403,7 +406,7 @@ impl QuoteStoring for Forget {
         &self,
         _: QuoteSearchParameters,
         _: DateTime<Utc>,
-        _: &QuoteSigningScheme,
+        _: QuoteKind,
     ) -> Result<Option<(QuoteId, QuoteData)>> {
         Ok(None)
     }
@@ -511,17 +514,7 @@ impl OrderQuoter {
             sell_token_price,
         };
 
-        let quote_kind = match parameters.signing_scheme {
-            QuoteSigningScheme::Eip1271 {
-                onchain_order: true,
-                ..
-            } => QuoteKind::Eip1271OnchainOrder,
-            QuoteSigningScheme::PreSign {
-                onchain_order: true,
-            } => QuoteKind::PreSignOnchainOrder,
-            _ => QuoteKind::Standard,
-        };
-
+        let quote_kind = quote_kind_from_signing_scheme(&parameters.signing_scheme);
         let quote = QuoteData {
             sell_token: parameters.sell_token,
             buy_token: parameters.buy_token,
@@ -624,11 +617,13 @@ impl OrderQuoting for OrderQuoter {
 
                     (id, data)
                 }
-                None => self
-                    .storage
-                    .find(parameters, now, signing_scheme)
-                    .await?
-                    .ok_or(FindQuoteError::NotFound(None))?,
+                None => {
+                    let quote_kind = quote_kind_from_signing_scheme(signing_scheme);
+                    self.storage
+                        .find(parameters, now, quote_kind)
+                        .await?
+                        .ok_or(FindQuoteError::NotFound(None))?
+                }
             };
             Ok(Quote::new(Some(id), data))
         };
@@ -682,6 +677,19 @@ impl From<&OrderQuoteRequest> for QuoteParameters {
     }
 }
 
+fn quote_kind_from_signing_scheme(scheme: &QuoteSigningScheme) -> QuoteKind {
+    match scheme {
+        QuoteSigningScheme::Eip1271 {
+            onchain_order: true,
+            ..
+        } => QuoteKind::Eip1271OnchainOrder,
+        QuoteSigningScheme::PreSign {
+            onchain_order: true,
+        } => QuoteKind::PreSignOnchainOrder,
+        _ => QuoteKind::Standard,
+    }
+}
+
 #[cfg(test)]
 mod tests {
     use super::*;
@@ -1403,12 +1411,11 @@ mod tests {
             from: H160([3; 20]),
             app_data: AppId([4; 32]),
         };
-        let signing_scheme = QuoteSigningScheme::Eip712;
 
         let mut storage = MockQuoteStoring::new();
         storage
             .expect_find()
-            .with(eq(parameters.clone()), eq(now), eq(signing_scheme))
+            .with(eq(parameters.clone()), eq(now), eq(QuoteKind::Standard))
             .returning(move |_, _, _| {
                 Ok(Some((
                     42,
```